_articles/front-proxy.md

```diff
@@ -0,0 +1,227 @@
+general-purpose control plane, as the Kubernetes ingress controller spec is
+lowest-common-denominator by design, with support for only a subset of Envoy's
+capabilities.
+
+Note that while running Envoy in Kubernetes is simpler if you’re committed to
+Kubernetes, many people start by running Envoy outside of Kubernetes in order
+to manage the migration. Running Envoy outside of Kubernetes ensures that any
+cluster failures doesn’t take down your Envoy front proxy. If you leave Envoy
+running outside the cluster after completing you migration, it also gives you
+flexibility to keep running other orchestrators, or even migrate to the thing
+that replaces Kubernetes!
+
+## SSL and Metrics
+
+Front proxies are a natural place to terminate SSL, to ensure that individual
+services don’t have to. You can either do in your cloud’s load balancer (e.g.
+AWS ELB) or in Envoy itself. To configure SSL, see [this article](ssl.html).
+
+As a common choke point for all traffic, front proxies are a great place to
+generate high-level metrics for your site. Make sure to at least send request
+health metrics to your dashboards: latency, volume, and success rate. Coarsely,
+this gives you a simple dashboard for “is the site up?” and partitioning the
+traffic by upstream cluster can give you health metrics for each service that
+are more relevant to the service teams.
+
+If you generate logs from your current front proxy, Envoy can be configured to
+send request-level access logs to a centralized server. Be aware that the
+volume of traffic at the edge can make logging expensive or slow, so this may
+not be cost-effective if it’s not a critical capability.
+
+Similarly, Envoy provides drop-in configuration for tracing, and and the front
+proxy is a great place to generate those tracing headers.
+
+## Multiple Data Centers Made Simple
+
+There are two strategies for running services in multiple zones (availability
+zones, regions, data centers, etc.). You can pick a single strategy, or you can
+run both hierarchically.
+
+If you have a _single network space_, your best bet is to use
+**Envoy’s Zone Aware Routing** to handle any underlying discrepancies in the
+topology seamlessly. This strategy is best used with closely connected zones
+like AWS Availability Zones, where multiple zones are operationally a single
+unit, but cross-zone traffic incurs a cost. In this setup, your network load
+balancer would balance across a front proxy that spans multiple zones, and
+Envoy would keep each request in the zone it was first load balanced to. This
+will minimize cross-zone traffic (which is expensive and slow), while
+maintaining the ability to incrementally fall back to out-of-zone instances if
+the local ones are failing.
+
+Setting up zone-aware load balancing requires two options to be set:
+
+  - Each cluster (statically defined or returned by the control plane via CDS) must enable zone-aware routing.
```

Comment:
Wrap, and also does this render correctly? Not sure about the space on the next line properly translating to a list.

Reply:
yea it works ok